src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -174,12 +174,20 @@ protected override async Task OnParametersSetAsync()
             return;
         }
 
-        if (PageViewModel.SelectedResource?.Name != _consoleLogsSubscription?.Name)
+        var selectedResourceName = PageViewModel.SelectedResource?.Name;
+        if (selectedResourceName != _consoleLogsSubscription?.Name)
         {
+            Logger.LogDebug("New resource {ResourceName} selected.", selectedResourceName);
+
             ConsoleLogsSubscription? newConsoleLogsSubscription = null;
-            if (PageViewModel.SelectedResource is not null)
+            if (selectedResourceName is not null)
             {
-                newConsoleLogsSubscription = new ConsoleLogsSubscription { Name = PageViewModel.SelectedResource!.Name };
+                newConsoleLogsSubscription = new ConsoleLogsSubscription { Name = selectedResourceName };
+                newConsoleLogsSubscription.CancellationToken.Register(state =>
+                {
+                    var s = (ConsoleLogsSubscription)state!;
+                    Logger.LogDebug("Canceling current subscription to {ResourceName}.", s.Name);
+                }, newConsoleLogsSubscription);
```

Comment:
This registration isn't needed if debug logging is not enabled. We can wrap this in `if (Logger.IsEnabled(LogLevel.Debug)) ...`.